core/org.eclipse.cdt.core.tests/model/org/eclipse/cdt/core/build/CBuildConfigurationManagerTests.java

```diff
@@ -139,6 +141,41 @@ public void testResetCachedScannerInfoProvider() throws Exception {
 				scannerInfoProvider instanceof ICBuildConfiguration);
 	}
 
+	/**
+	 * Test {@link CBuildConfigurationManager#getBuildConfiguration(IBuildConfiguration buildConfig)}
+	 * IBuildConfigurations with a valid ICBuildConfiguration pendant will be removed from
+	 * the CBuildConfigurationManager.noConfigs Set.
+	 */
+	@Test
+	public void testNoConfigs() throws Exception {
+		// create a CMake project; performed in setup()
+
+		// Create a new IBuildConfiguration.
+		CMakeBuildConfigurationProvider provider = new CMakeBuildConfigurationProvider();
+		String buildConfigBaseName = "testNoConfigs";
+		IBuildConfiguration buildConfiguration = configManager.createBuildConfiguration(provider, project,
+				buildConfigBaseName, new NullProgressMonitor());
+
+		// Create a new ICBuildConfiguration.
+		String buildConfigName = provider.getId() + "/" + buildConfigBaseName;
+		ILaunchTarget launchTarget = launchTargetManager.getLocalLaunchTarget();
+		ICBuildConfiguration cBuildConfig = new CMakeBuildConfiguration(buildConfiguration, buildConfigName,
+				mockToolchain, null, ILaunchManager.RUN_MODE, launchTarget);
+
+		// Try to get ICBuildConfiguration, before the IBuildConfiguration/ICBuildConfiguration pair was
+		// added to the build configuration manager. Such a getAdapter() call could happen in a parallel
+		// process. It will fail and buildConfiguration will be placed in configManager.noConfigs
+		ICBuildConfiguration cbConfig = buildConfiguration.getAdapter(ICBuildConfiguration.class);
+		assertThat("bConfig is expect to be null", cbConfig == null);
+
+		// Register the IBuildConfiguration/ICBuildConfiguration pair.
+		configManager.addBuildConfiguration(buildConfiguration, cBuildConfig);
+
+		// Get the ICBuildconfiguration pendant
+		cbConfig = buildConfiguration.getAdapter(ICBuildConfiguration.class);
+		assertThat("A valid ICBuildconfiguration counterpart is found", cbConfig == cBuildConfig);
```

Comment:
I would replace
		assertThat("A valid ICBuildconfiguration counterpart is found", cbConfig == cBuildConfig);
with
		assertThat(cbConfig, equalTo(cBuildConfig));